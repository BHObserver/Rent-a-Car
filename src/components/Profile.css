.car-list {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
    gap: 20px;
  }
  
  .car-item {
    background-color: #f9f9f9;
    border-radius: 5px;
    padding: 15px;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  }
  
  .car-photo {
    width: 100%;
    border-radius: 5px;
    margin-bottom: 10px;
  }
  
  .car-name {
    font-size: 18px;
    font-weight: bold;
    margin-bottom: 5px;
  }
  
  .car-model {
    font-size: 16px;
    color: #666;
    margin-bottom: 5px;
  }
  
  .car-details {
    font-size: 14px;
    color: #444;
    margin-bottom: 5px;
  }
  
  .car-city,
  .car-cost {
    font-size: 14px;
    color: #888;
  }
  