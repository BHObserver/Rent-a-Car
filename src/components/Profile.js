import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import fetchCars from '../redux/actions/carActions';
import '../App.css';

const Profile = () => {
  const dispatch = useDispatch();
  const { cars, loading, error } = useSelector((state) => state.car) || {};

  useEffect(() => {
    dispatch(fetchCars());
  }, [dispatch]);

  return (
    <div>
      <h2>Car List</h2>
      {loading && <div>Loading...</div>}
      {error && (
        <div>
          Error:
          {error}
        </div>
      )}
      {cars && cars.length > 0 && (
        <div className="car-list">
          {cars.map((car) => (
            <div key={car.id} className="car-item">
              <img src={car.photo} alt={car.name} className="car-photo" />
              <div className="car-details">
                <div className="car-name">{car.name}</div>
                <div className="car-model">{car.model}</div>
                <div className="car-details">{car.details}</div>
                <div className="car-city">
                  City:
                  {car.city}
                </div>
                <div className="car-cost">
                  Cost:
                  {car.cost}
                </div>
              </div>
            </div>
          ))}
        </div>
      )}
    </div>
  );
};

export default Profile;
