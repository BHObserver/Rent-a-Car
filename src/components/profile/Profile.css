.profile-container {
  margin: 20px;
}

.heading {
  color: #fff;
}

.car-list-container {
  position: relative;
}

.carousel-container {
  position: relative;
}

.prev-btn {
  position: absolute;
  top: 50%;
  left: 10px;
  transform: translateY(-50%);
  background-color: transparent;
  color: #000;
  border: none;
  font-size: 24px;
  cursor: pointer;
  outline: none;
}

.next-btn {
  position: absolute;
  top: 50%;
  right: 10px;
  transform: translateY(-50%);
  background-color: transparent;
  color: #c79a12cd;
  border: none;
  font-size: 24px;
  cursor: pointer;
  outline: none;
}

.car-list {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  gap: 20px;
}

.car-item {
  background-color: #f9f9f9;
  border-radius: 5px;
  padding: 15px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}

.car-photo {
  width: 100%;
  border-radius: 5px;
  margin-bottom: 10px;
}

.car-name {
  font-size: 18px;
  font-weight: bold;
  margin-bottom: 5px;
}

.car-model {
  font-size: 16px;
  color: #666;
  margin-bottom: 5px;
}

.car-details {
  font-size: 14px;
  color: #444;
  margin-bottom: 5px;
}

.car-city,
.car-cost {
  font-size: 14px;
  color: #888;
}
