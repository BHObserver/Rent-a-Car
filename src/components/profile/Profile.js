import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { CircularProgress, Typography } from '@mui/material';
import { styled } from '@mui/system';
import { useNavigate } from 'react-router-dom';
import { fetchCars } from '../../redux/actions/carActions';
import CarCarousel from '../car/CarCarousel';
import CarCard from '../car/CarCard';
import './Profile.css';

const Container = styled('div')({
  textAlign: 'center',
  padding: '20px',
});

const Spinner = styled(CircularProgress)({
  margin: '20px',
});

const ErrorText = styled(Typography)({
  margin: '20px',
});

const Profile = () => {
  const dispatch = useDispatch();
  const { cars, loading, error } = useSelector((state) => state.car.cars);
  const [currentIndex, setCurrentIndex] = useState(0);
  const navigate = useNavigate();

  useEffect(() => {
    dispatch(fetchCars());
  }, [dispatch]);

  const handlePrev = async () => {
    setCurrentIndex((prevIndex) => Math.max(0, prevIndex - 1));
  };

  const handleNext = async () => {
    setCurrentIndex((prevIndex) => Math.min(cars.length - 1, prevIndex + 1));
  };

  const handleCardClick = (carId) => {
    navigate(`/cars/${carId}`); // Use navigate to navigate to the car details page
  };

  return (
    <Container>
      <Typography variant="h3" gutterBottom>
        <h3 className="heading"> List of Cars </h3>
      </Typography>
      {loading && <Spinner />}
      {error && (
        <ErrorText variant="body1" color="error">
          Error:
          {' '}
          {error}
        </ErrorText>
      )}
      {/* Check if cars is defined before mapping */}
      {cars && cars.length > 0 && (
        <CarCarousel
          cars={cars}
          currentIndex={currentIndex}
          handlePrev={handlePrev}
          handleNext={handleNext}
          loading={loading}
        >
          {/* Render CarCard inside CarCarousel */}
          {cars.map((car) => (
            <CarCard key={car.id} car={car} onCardClick={handleCardClick} />
          ))}
        </CarCarousel>
      )}
    </Container>
  );
};

export default Profile;
