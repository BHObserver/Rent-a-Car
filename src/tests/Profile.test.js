import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Profile from '../components/Profile';

const mockStore = configureStore([]);

describe('Profile component', () => {
  test('renders correctly', () => {
    // Mock the Redux state
    const initialState = {
      car: {
        cars: [],
        loading: false,
        error: null,
      },
    };

    // Create a mock store instance
    const store = mockStore(initialState);

    const tree = renderer.create(
      <Router>
        {' '}
        {/* Wrap Profile with BrowserRouter */}
        <Provider store={store}>
          <Profile />
        </Provider>
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
