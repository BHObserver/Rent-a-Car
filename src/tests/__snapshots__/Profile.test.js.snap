// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component renders correctly 1`] = `
Array [
  " ",
  <div
    className="css-1nttawi"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
    >
      <h3
        className="heading"
      >
         List of Cars 
      </h3>
    </h4>
  </div>,
]
`;
